### ISSUE TITLE
A false negative caused by support of lamda expression

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3442]
**Affects PMD Version:6.35.0** 

**Rule: AvoidInstantiatingObjectsInLoops**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd-6.36.0/pmd_rules_java_performance.html#avoidinstantiatingobjectsinloops>

**Description:**
This is a false negative. I found an object initialized in the loop and no warnings. This loop is generated by lamda expression. I think this can help improve PMD's soundness. Thanks for your consideration!

**Code Sample demonstrating the issue:**

```java
public void test() {
    ArrayList<Obejct> strs = new ArrayList<>();
    strs.forEach(str->{
        Foo foo = new Foo();  // should report a warning here
    });
}
```

**Expected outcome:A warning**

PMD should report a violation at line 4, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by Zustin** [comment]
@oowekyala, Hi clement, could you please help check this issue? I think it is a false negative and am willing to help you repair it. Thanks.
